### PR TITLE
[SPARK-6635][SQL] DataFrame.withColumn should replace columns with identical column names

### PR DESCRIPTION
**Pull request by viirya**
JIRA https://issues.apache.org/jira/browse/SPARK-6635

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala [hunk 1]
```diff
@@ -457,6 +457,10 @@ class DataFrameSuite extends QueryTest {
         Row(key, value, key + 1)
       }.toSeq)
     assert(df.schema.map(_.name).toSeq === Seq("key", "value", "newCol"))
+
+    val df2 = TestSQLContext.sparkContext.parallelize(Array(1, 2, 3)).toDF("x")
+    val df3 = df2.withColumn("x", df2("x") + 1)
+    assert(df3.select("x").collect().toSeq === Seq(Row(2), Row(3), Row(4)))
```
Comment (marmbrus) [inline sql/core/src/test/scala/org/apache/spark/sql/DataFrameSuite.scala]: Make this its own test and use `checkAnswer` to check the result.

**Comment by marmbrus** [comment]
Thanks!  Merged to master.

**Comment by bplaster** [comment]
This should probably be noted here: https://spark.apache.org/docs/latest/api/scala/index.html#org.apache.spark.sql.DataFrame

**Comment by marmbrus** [comment]
Pull requests to improve documentation are always welcome :)

**Pull request closed without merging**